components/raftstore/src/store/peer_storage.rs

```diff
@@ -436,16 +438,42 @@ where
     /// Gets a snapshot. Returns `SnapshotTemporarilyUnavailable` if there is no
     /// unavailable snapshot.
     pub fn snapshot(&self, request_index: u64, to: u64) -> raft::Result<Snapshot> {
+        let peer = find_peer_by_id(&self.region, self.peer_id).unwrap();
```

Comment:
How about setting a flag instead of searching?

Reply:
done